### PR TITLE
fix: fixed conflicts and repushed Endpoint to Get specific product details by ID by Superadmin

### PR DESCRIPTION
**Pull request by froschi95**
​

## Description

Implement a new API endpoint for retrieving product details, accessible only to superadmins. This endpoint will be part of the dashboard functionality.

#### Endpoint Details:
Path: GET /api/v1/dashboard/products/{product_id}
Access: Restricted to superadmins only

Response:
```json
{
  "status_code": 200,
  "message": "Product retrieved successfully",
  "data": {
    "id": "string",
    "name": "string",
    "description": "string",
    "price": 0.00,
    "org_id": "string"
  }
}
```

​
## Related Issue (#449 )
This PR is also related to my first PR(#466) which I had to close in order to properly resolve conflicts.

​

## Motivation and Context

​

## How Has This Been Tested?

- Unit tests were written sing pytest module to cover
 - Successful retrieval by a superadmin
 - Attempt to access by a non-superadmin user
 - Attempt to retrieve a non-existent product

​

## Screenshots (if appropriate - Postman, etc):

​

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
      ​

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Resolve conflicts

**Comment by froschi95** [comment]
> Resolve conflicts

conflicts resolved

**Comment by froschi95** [comment]
@joboy-dev

File: api/v1/routes/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    # Check if the user is a superadmin
+    if not current_user.is_super_admin:
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="You are NOT authorized to access this endpoint"
+        )
+    
+    product = product_service.fetch(db, product_id)
+    if product is None:
```
Comment (joboy-dev) [inline api/v1/routes/dashboard.py:46]: No need for this check

File: api/v1/routes/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    Returns:
+        ProductResponse: A response containing the product data, status code, and a message.
+
+    Raises:
+        HTTPException: 
+            - 403 error if the user is not a superadmin.
+            - 404 error if the product is not found.
+    """
+    # Check if the user is a superadmin
+    if not current_user.is_super_admin:
```
Comment (joboy-dev) [inline api/v1/routes/dashboard.py:39]: No need for this check

File: api/v1/routes/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        )
+    
+    product = product_service.fetch(db, product_id)
+    if product is None:
+        raise HTTPException(
+            status_code=status.HTTP_404_NOT_FOUND,
+            detail=f"Product with id {product_id} not found"
+        )
+    
+    return DashboardProductResponse(
```
Comment (joboy-dev) [inline api/v1/routes/dashboard.py:52]: Use success_response() function, already in the codebase. You can check other files for how to use it

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Pull recent changes and update your code. Thanks

**Pull request closed without merging**